presto-main/src/main/java/com/facebook/presto/operator/project/PageProcessor.java

```diff
@@ -116,7 +116,8 @@ private static boolean isUnloadedLazyBlock(Block block)
     private static long calculateRetainedSizeWithoutLoading(Page page)
     {
         long retainedSizeInBytes = 0;
-        for (Block block : page.getBlocks()) {
+        for (int channel = 0; channel < page.getChannelCount(); channel++) {
```

Comment:
Would it make sense to have a method that returns an `Iterable<Block>` so that for-each loops work?

Reply:
I was thinking we might want that, or a `stream()` method, or maybe a `forEachBlock` method, but there are currently only thee users of iteration without index, so I decided to delay the decision for now.

Reply:
`stream()` is cool. I am, however, concerned about streams overhead in execution. I didn't do my own measurement though.